loadData/person_data.rb

```diff
@@ -0,0 +1,24 @@
+require 'json'
+require_relative '../school/student'
+require_relative '../school/teacher'
+
+@file_person = File.read('./data/person.json')
+
+def person_data(person)
+  return if @file_person.empty?
+
+  data_person = JSON.parse(@file_person)
+  data_person.map do |value|
+    name = value['name']
+    age = value['age']
+    if value['classroom'].nil?
+      specialization = value['specialization']
+      person_data = Teacher.new(specialization, age, name)
+    else
+      classroom = value['classroom']
+      person_data = Student.new(classroom, age, name)
+    end
+    person_data.id = value['id']
+    person << person_data
+  end
+end
```

Comment:
![image](https://user-images.githubusercontent.com/65068771/176337871-2f552343-f936-4387-8fcc-fe260e226529.png)

- [ ] Great job on saving data in JSON files, However, in the requirements it said that you should handle the case where one of the JSON files is missing the application should not throw an error. I deleted the `/data/person.json` file and the app crashed. Check this hint solution :point_down: 

```rb
# Kindly note that this is just an example solution, feel free to solve it the way you want, just make sure the app does not  crash when this file is missing

require 'json'
require_relative '../school/student'
require_relative '../school/teacher'


def person_data(person)
  File.write('./data/person.json', JSON.pretty_generate([])) unless File.exists?('./data/person.json')
  @file_person = File.read('./data/person.json')
  return if @file_person.empty?

  data_person = JSON.parse(@file_person)
  data_person.map do |value|
    name = value['name']
    age = value['age']
    if value['classroom'].nil?
      specialization = value['specialization']
      person_data = Teacher.new(specialization, age, name)
    else
      classroom = value['classroom']
      person_data = Student.new(classroom, age, name)
    end
    person_data.id = value['id']
    person << person_data
  end
end